core/src/test/scala/com/microsoft/azure/synapse/ml/causal/VerifyOrthoDMLEstimator.scala

```diff
@@ -0,0 +1,117 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.causal
+
+import com.microsoft.azure.synapse.ml.core.test.fuzzing.{EstimatorFuzzing, TestObject}
+import org.apache.spark.ml._
+import org.apache.spark.ml.feature.VectorAssembler
+import org.apache.spark.ml.util.MLReadable
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.{DoubleType, StringType, StructField, StructType}
+import org.scalactic.Equality
+
+class VerifyOrthoDMLEstimator extends EstimatorFuzzing[OrthoForestDMLEstimator] {
+    val schema = StructType(Array(
+      StructField("Label",StringType),
+      StructField("X1",DoubleType),
+      StructField("X2",DoubleType),
+      StructField("X3",DoubleType),
+      StructField("W1",DoubleType),
+      StructField("W2",DoubleType),
+      StructField("W3",DoubleType),
+      StructField("W4",DoubleType),
+      StructField("W5",DoubleType),
+      StructField("W6",DoubleType),
+      StructField("W7",DoubleType),
+      StructField("W8",DoubleType),
+      StructField("W9",DoubleType),
+      StructField("W10",DoubleType),
+      StructField("Y",DoubleType),
+      StructField("T",DoubleType),
+      StructField("TE",DoubleType)
+    ))
+    val treatmentCol = "T"
+    val outcomeCol = "Y"
+
+    val heterogeneityCols  = Array("X1","X2","X3")
+    val heterogeneityVecCol = "XVec"
+    val confounderCols = Array("W1","W2","W3","W4","W5","W6","W7","W8","W9","W10")
+    val confounderVecCol = "XWVec"
+
+    val filePath = s"${sys.env("DATASETS_HOME")}/Causal/OrthoForestData.csv"
+
+    val df = spark.read.format("csv")
+      .option("header", "true")
+      .schema(schema)
+      .load(filePath)
+
```

Comment:
does inferSchema not work here?

Reply:
Unfortunately, not, the Target variable gets it as int. Just want to make the test reliable